### PR TITLE
Add the CONTRIBUTING.md files

### LINKED ISSUE #3: Create the CONTRIBUTION.md
**Short description:**
Create the CONTRIBUTION.md file that explains the contribution rules.

**Describe the solution you'd like**
This file should clarify the rules concerning:
- Issues
- Branches
- Coding style
- Documentation
- Committing style
- Testing policy
- Pull resquets

### PR DESCRIPTION
**Pull request by ZQUEMA**
# Purpose:
Add the file CONTRIBUTING.md to declare the rules for contributing to this project.

## Added features:
The file CONTRIBUTING.md is added. This file clarifies the rules concerning:
- Issues
- Branches
- Coding style
- Documentation
- Committing style
- Testing policy
- Pull requets

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
+# Development Rules
+
+The rules to contribute to the R-Type project are described below.
```
Comment (LucasTesnier) [inline CONTRIBUTING.md]: R-Type project, really ?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+`Create the Contributing.md`
+
+* Describing the issue is mandatory and usefull for everyone. Try to be concise with the features which will be added on the project.
+
+* Add a label to clarify the issue. If there is no label existing or corresponding to your issue, create one and describe it.
+
+* Take a moment to link the issue with the project, if it hasn't been done automatically.
+
+* If possible, assign someone to it.
+
```
Comment (LucasTesnier) [inline CONTRIBUTING.md:24]: Add a part about DueDate for issue

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+Try to push as regularly as possible so everyone can see where you are at.
+When first committing on a branch, it is recommended to open a PR draft, so it will be easier for your mates to see where you are at.
+
+## Coding Style
+
+For the coding style, there are some rules you need to follow:
+
+* Everybody working on the project with you must be able to understand on what you are working within minutes. If you can split a function in two, you must do it !
+
+* Your code must respect the Clang format. It's necessary to be able to compile on Windows. While coding, press at the same time CTRL+MAJ+I to apply the format !
```
Comment (LucasTesnier) [inline CONTRIBUTING.md]: Need to be actualized.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+
+## Documentation Rules
+
+We use JSDoc 3 for documentation, and in order to allow the proper deployment of the documentation, please respect the JSDoc format. <br>
+If you have a doubt about the format, you can click [here](https://jsdoc.app/howto-commonjs-modules.html).
+
```
Comment (LucasTesnier) [inline CONTRIBUTING.md:52]: Where's the part about Flutter documentation ?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+Any undocumented elements will lead to the refusal of the PR.
+
+## Committing Rules
+
+First, the commits. They should always look like the following description:
```
Comment (LucasTesnier) [inline CONTRIBUTING.md]: First ?

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+
+You should always commit frequently and based on the given exemple.
+
+If possible, please certify your commits (-S).
+
+## Testing Rules
+
+When something can be verified by a test, we will be looking for tests in your PR.
+
+To compile your tests, the corresponding CMakeLists.txt should look like this:
```
Comment (LucasTesnier) [inline CONTRIBUTING.md]: Need to be removed.
|
v

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+    #include <criterion/criterion.h>
+
+    /// @brief Very Basic Test
+    Test(Basic, some_basics_tests)
+    {
+        cr_assert_eq(42, 42);
+    }
+```
+
+For more details, you can go from the project's root to tests/unit_tests directory.
```
Comment (LucasTesnier) [inline CONTRIBUTING.md]: ^
|

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+
+If you're in functionnal tests, please add a bash/bat script to execute them.
+
+The lack of test in your code can be one of the reasons of refusing a PR.
+
+## Pull Request Rules
+
+### The creator
```
Comment (LucasTesnier) [inline CONTRIBUTING.md]: The creator ? Owner / Author ?

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+
+For the reviewers, prioritize the people who can really help you with your work.
+
+A reviewer must read, understand and test
+
+You must have at least 2 reviewers who approve your pull request.
+
+Try not to exceed 24 hours before reading your PRs.
+
+It is obviously forbidden to approve a merge until all the reviewers have accepted the PR as well as the success of the workflows.
```
Comment (LucasTesnier) [inline CONTRIBUTING.md:126]: Maybe add a conclusion part with ours contact ?

**Review by LucasTesnier — CHANGES_REQUESTED** [review]
Some things to change / Question to answer

File: CONTRIBUTING.md (hunk 1 above)
Comment (ZQUEMA) [reply CONTRIBUTING.md]: Modification done

File: CONTRIBUTING.md (hunk 9 above)
Comment (ZQUEMA) [reply CONTRIBUTING.md:126]: Will be done in the README.md

**Review by LucasTesnier — APPROVED** [review]

**Review by GuyomT — APPROVED** [review]
Agreed w/ contributing guidelines

Guigui

**Review by tbellicha — APPROVED** [review]

**Review by TomDUVAL-MAHE — APPROVED** [review]

**Pull request merged**